specification/orbital/resource-manager/Microsoft.Orbital/stable/2022-11-01/orbital.json

```diff
@@ -0,0 +1,2223 @@
+          "description": "Spacecraft elevation above the horizon at contact start."
+        },
+        "endElevationDegrees": {
+          "type": "number",
+          "readOnly": true,
+          "description": "Spacecraft elevation above the horizon at contact end."
+        }
+      },
+      "description": "Contact Instance Properties."
+    },
+    "Contact": {
+      "type": "object",
+      "required": [
+        "properties"
+      ],
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/ContactsProperties"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ProxyResource"
+        }
+      ],
+      "description": "Customer creates a contact resource for a spacecraft resource."
+    },
+    "AvailableContactsListResult": {
+      "type": "object",
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/AvailableContacts"
+          },
+          "x-ms-identifiers": [],
+          "description": "A list of available contacts."
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "The URL to get the next set of results.",
+          "readOnly": true
+        }
+      },
+      "description": "Response for the ListAvailableContacts API service call."
+    },
+    "AvailableContacts": {
+      "type": "object",
+      "properties": {
+        "spacecraft": {
+          "allOf": [
+            {
+              "$ref": "#/definitions/ResourceReference"
+            }
+          ],
+          "description": "The reference to the spacecraft resource."
+        },
+        "groundStationName": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Name of Azure Ground Station."
+        },
+        "properties": {
+          "x-ms-client-flatten": true,
+          "allOf": [
+            {
+              "$ref": "#/definitions/ContactInstanceProperties"
+            }
+          ],
+          "description": "Properties of Contact resource."
+        }
+      },
+      "description": "Customer retrieves list of Available Contacts for a spacecraft resource. Later, one of the available contact can be selected to create a contact."
+    },
+    "AvailableGroundStationListResult": {
+      "type": "object",
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/AvailableGroundStation"
+          },
+          "description": "A list of ground station resources."
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "The URL to get the next set of results.",
+          "readOnly": true
+        }
+      },
+      "description": "Response for the AvailableGroundStations API service call."
+    },
+    "AvailableGroundStation": {
```

Comment:
>AvailableGroundStation" [](http://example.com/codeflow?start=5&length=23)

Why not simply derive the AvailableGroundStation from common-types ProxyResource?

Reply:
we tried doing that, but it throws a swgagger lint diff error because it detects it as tracked resource which might be due to the location attribute imo

Reply:
That would make sense. Normally only TrackedResource has location. I'm a bit surprised actually that you are not deriving from TrackedResource now that you mention it. 
Was there a special reason that you chose ProxyResource and not TrackedResoruce?
It seems you don't have Tags, so far, but Tags seems like a feature you should want to support!

Reply:
I guess its because they are immutable / read only ?

Reply:
I think inheriting from ProxyResource, and declaring the aditional location attribute is still the right thing to do, what was the error you got? Can you suppress it?

Reply:
Bump - I still don't see any reason not to inherit ProxyResource. We are trying to move in the direction of every resource type inheriting either ProxyResource or TrackedResoruce for consistency. Its still kinda optional but if you don't fix it now, you will probably be bugged about fixing it later.